qiskit/_result.py

```diff
@@ -253,4 +318,88 @@ def get_qubitpol_vs_xval(self, xvals_dict=None):
             for qubit_ind in range(nqubits):
                 qubitpol[circuit_ind,qubit_ind] = self.average_data(self.__qobj['circuits'][circuit_ind]['name'], z_dicts[qubit_ind])
 
-        return qubitpol,xvals
\ No newline at end of file
+        return qubitpol,xvals
+
+    def save_datestr(self, folder, fileroot):
+        """Constructs a filename using the current date-time
+
+        Args:
+            folder: path to the save folder
+            fileroot: root string for the file
+
+        Returns:
+            filename: full file path of the form 'folder/YYYY_MM_DD_HH_MM_fileroot.json'
+        """
+
+        #if the fileroot has .json appended strip it off
+        if (len(fileroot)>4 and fileroot[-5:].lower()=='.json'):
+            fileroot = fileroot[0:-5]
+
+        return os.path.join(folder,'{:%Y_%m_%d_%H_%M_}'.format(datetime.datetime.now())+fileroot+'.json')
+
+    def save(self, filename, metadata=None):
+        """Save a result (qobj + result) to a single dictionary file filename.json
+        If the file already exists then numbers will be appended to generate a unique filename
+
+        Args:
+            filename: save path
+            metadata (optional): Add another dictionary with custom data for the result (eg fit results)
+
+        Returns:
+            filename: full file path
+        """
+        master_dict = {}
+        master_dict['qobj'] = self.__qobj.copy()
+        master_dict['result'] = self.__result.copy()
+        if metadata is None:
+            master_dict['metadata'] = {}
+        else:
+            master_dict['metadata'] = metadata
+
+
+        #need to convert any ndarray variables to lists so that they can be
+        #exported to the json file
+        self._convert_qobj_to_json(master_dict['result'])
+
+        #if the filename has .json appended strip it off
+        if filename[-5:].lower()=='.json':
```

Comment:
In the spirit of the comment for `save_datestr()`, would it be possible to assume that the user is responsible for determining the filename and avoid trying to extract the `.json` part?